### PR TITLE
ISPN-13602 Add trace logs to show server http requests and responses

### PR DESCRIPTION
**Pull request by ryanemerson**
https://issues.redhat.com/browse/ISPN-13602

### CODE REVIEW

File: server/rest/src/main/java/org/infinispan/rest/HttpMessageUtil.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+      return sb.toString();
+   }
+
+   private static void appendCommon(StringBuilder buf, HttpMessage msg) {
+      buf.append(StringUtil.simpleClassName(msg));
+      buf.append("(decodeResult: ");
+      buf.append(msg.decoderResult());
+      buf.append(", version: ");
+      buf.append(msg.protocolVersion());
+      buf.append(')');
```
Comment (danberindei) [inline server/rest/src/main/java/org/infinispan/rest/HttpMessageUtil.java:40]: I'm not sure any of these are interesting enough to write in the log at all, not to mention in a line by itself.
I would skip it and start directly with `appendInitialLine()`.

File: server/rest/src/main/java/org/infinispan/rest/HttpMessageUtil.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+      buf.append(res.status());
+      buf.append(StringUtil.NEWLINE);
+   }
+
+   private static void appendHeaders(StringBuilder buf, HttpHeaders headers) {
+      for (Map.Entry<String, String> header : headers) {
+         buf.append(header.getKey());
+         buf.append(": ");
+         buf.append(header.getValue());
+         buf.append(StringUtil.NEWLINE);
```
Comment (danberindei) [inline server/rest/src/main/java/org/infinispan/rest/HttpMessageUtil.java:65]: We could write the newline first, then we wouldn't need to remove the last newline at the end.

**Review by danberindei — CHANGES_REQUESTED** [review]

**Comment by ryanemerson** [comment]
@danberindei Updated

**Comment by danberindei** [comment]
Merged, thanks Ryan!

**Pull request merged**